### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by codecaiine**
- I created model classes for all entities as shown in the ERD diagram.
- I set up associations between models.
- I made current user data available to all controllers.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  User.first
 end
```
Comment (rukundoeric) [inline app/controllers/application_controller.rb]: - [ ] <b>REQUIRED:</b> Project requires to create the function `current_user` that will return the first user. Kindly to it to comply with project requiremenst.

```ruby
def current_user
  User.first
end

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @codecaiine, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] Check the comments under the review.

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (codecaiine) [reply app/controllers/application_controller.rb]: Hi @rukundoeric  thank you for reviewing my work and also your comments. I fixed it.

**Review by Lameck1 — APPROVED** [review]
Hi, @codecaiine,

You've done addressing previous review concerns 👏🏼.
The tasks for this milestone are now successfully implemented 💯.
All the best with the next milestone.

You are free to merge this branch :shipit:

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by codecaiine** [comment]
Hi @Lameck1 ,

Thank you for reviewing and approving my work. 

Cheers!

**Pull request merged**